examples/TestRoutingGuard/can-load.spec.ts

```diff
@@ -0,0 +1,172 @@
+import { Location } from '@angular/common';
+import {
+  Component,
+  inject,
+  Injectable,
+  NgModule,
+  VERSION,
+} from '@angular/core';
+import {
+  CanLoadFn,
+  Router,
+  RouterModule,
+  RouterOutlet,
+} from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { from } from 'rxjs';
+import { mapTo } from 'rxjs/operators';
+
+import {
+  MockBuilder,
+  MockRender,
+  NG_MOCKS_GUARDS,
+  NG_MOCKS_ROOT_PROVIDERS,
+  ngMocks,
+} from 'ng-mocks';
+
+// A simple service simulating login check.
+// It will be replaced with its mock copy.
+@Injectable()
+class LoginService {
+  public isLoggedIn = false;
+}
+
+// The canLoad guard we want to test.
+const canLoadGuard: CanLoadFn = (route, segments) => {
+  if (route && segments && inject(LoginService).isLoggedIn) {
+    return true;
+  }
+
+  return from(inject(Router).navigate(['/login'])).pipe(mapTo(false));
+};
+
+// Another canLoad guard like in a real world example,
+// which should be removed from testing to avoid side effects on the route.
+const sideEffectCanLoadGuard: CanLoadFn = () => false;
+
+// A simple component pretending a login form.
```

Comment:
// A simple component pretending a login form. -> // A simple component pretending **to be** a login form.